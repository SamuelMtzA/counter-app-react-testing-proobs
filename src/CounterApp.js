//rafcp
import React, {useState} from 'react'
import PropTypes from 'prop-types'
//props son comunicacion entre componentes
//value es de properties, y counterApp es un componente funcional
const CounterApp = ({value = 10}) => {
    //hooks useState es una funcion que nos permite crear estados en nuestros componentes funcionales
    const [counter, setCount] = useState( value );

    //handleAdd
    const handleAdd = () => {
        //no se puede mutar el state con conuter ++
        setCount(counter + 1)
    }

    const handleDis= () =>{
        setCount(counter - 1)
    }

    const handleRes= () =>{
        setCount(value)
    }
    return (
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            {/* pasar funcion como referencia */}
            <button onClick = { handleAdd }>+1</button>
            <button onClick = { handleRes }>reset</button>
            <button onClick = { handleDis }>-1</button>

        </>
    )
}

CounterApp.propTypes = {
 value : PropTypes.number
}

export default CounterApp
