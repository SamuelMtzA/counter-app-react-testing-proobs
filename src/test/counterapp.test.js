//pruebas automatizadas realizadas para que la aplicacion se comporte como nosotros esperamos
import React from "react";
import { shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe("pruebas en counter app", () => {
  //esta instruccion es para que el wrapper en cada una de las pruebas se pueda modificar sin que se pierda el estado
  let wrapper = shallow(<CounterApp />);
  beforeEach(() => {
    wrapper = shallow(<CounterApp />);
  });

  test("debe de mostrar <counterapp /> correctament, hacer match con snap", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de mostrar <counterapp /> el valor por defecto de 100", () => {
    const valor = 100;
    const wrapper = shallow(<CounterApp value={valor} />);
    //trim para borrar espacios en blanco
    const counterText = wrapper.find("h2").text().trim();
    // console.log(counterText);
    expect(counterText).toBe("100");
  });

  //simulacion de un evento click
  test("debe de incrementar el valor en 1 con el boton", () => {
    //simular el evento click
    wrapper.find("button").at(0).simulate("click");
    // console.log(btn1);
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("11");
  });

  test("debe de decrementar el valor en 1 con el boton", () => {
    //simular el evento click, se queda almacenado el valor anterior
    wrapper.find("button").at(2).simulate("click");
    // console.log(btn1);
    const counterText = wrapper.find("h2").text().trim();

    expect(counterText).toBe("9");
  });

  test("debe de colocar el valor por defecto de reset", () => {
    const wrapper = shallow(<CounterApp value={105} />);
    //realizar el evento para poder simular el reset
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");


    const counterText = wrapper.find("h2").text().trim();
    // console.log(counterText);
    expect(counterText).toBe("105");

  })
});
