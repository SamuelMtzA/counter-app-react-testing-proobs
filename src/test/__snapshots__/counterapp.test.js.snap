// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en counter app debe de mostrar <counterapp /> correctament, hacer match con snap 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
